src/fsharp/ConstraintSolver.fs

```diff
@@ -2575,11 +2575,21 @@ let CodegenWitnessThatTypSupportsTraitConstraint tcVal g amap m (traitInfo:Trait
                 | true, true, 1 -> 
                         Some (mkStaticRecdFieldSet (rfref, tinst, argExprs.[0], m))
                 | true, false, 2 -> 
-                        Some (mkRecdFieldSet g (argExprs.[0], rfref, tinst, argExprs.[1], m))
```

Comment:
@TIHan This is another of the cases - it is when a trait constraint solves to be a record field setter